### PR TITLE
Make Modals a little prettier

### PR DESCRIPTION
**Pull request by MagdalenaHenke**
Modals aren't fully defined yet, so there isn't a real spec for them. But we do have them. And I know for example Spacer uses them and had to do some overrides to make em prettier. 

So I figured I'd just make em a liiiittle bit prettier. Borrowing from the current [Alerts](https://app.zeplin.io/project/5adf8fe14c3924b93d974cba/screen/5b43ae0ddd03f30b025bd725) spec.

**Before**
Screen Shot 2019-06-18 at 5 28 03 PM
Screen Shot 2019-06-18 at 5 28 07 PM

**After**
Screen Shot 2019-06-18 at 5 27 47 PM
Screen Shot 2019-06-18 at 5 27 43 PM

### CODE REVIEW

File: packages/blocks-base/styles/buttons.styl [hunk 1]
```diff
@@ -259,15 +259,16 @@ a.blx-button {
 .blx-h-button-group {
```
Comment (MagdalenaHenke) [inline packages/blocks-base/styles/buttons.styl]: definitely look at this one without whitespace changes - my sublime cuts off unnecessary white space, the actual meaty div is a lot smaller

File: packages/blocks-react/components/modals/BaseModal.jsx [hunk 2]
```diff
@@ -17,10 +18,9 @@ const BaseModal = (props) => {
       <div className="blx-modal">
         {
           props.isClosable && (
-            <button
-              className="blx-icon blx-icon-close"
-              onClick={props.onClose}
-            />
+            <button className="blx-modal-close" onClick={props.onClose}>
+              <CloseIcon />
+            </button>
```
Comment (MagdalenaHenke) [inline packages/blocks-react/components/modals/BaseModal.jsx]: making it similar to the close button for `Alerts` so I could add a hover style

**Comment by MagdalenaHenke** [comment]
@emma-floored @Jinsoo94 this one is good for review

**Comment by emma-floored** [comment]
👍

**Pull request merged**